### ISSUE TITLE
Copy & paste by ctrl c / ctrl v

### ISSUE THREAD

**Issue by jhuda1** [cedrozor/myrtille#346]
Has anyone implemented the option of copying and pasting from the computer to the browser and vice versa, by combining the keys, I think it is something that will make it very easy for the users and it is important.
I would love to hear from someone who has done this. The way I checked in the meantime it does not work.

**Comment by cedrozor** [comment]
Hi, did you used https for the page ? copy & paste requires it, in addition to allow clipboard access when prompted

**Comment by support-tt** [comment]
> Hi, did you used https for the page ? copy & paste requires it, in addition to allow clipboard access when prompted

Yes I can confirm that this is working with https and real certificate (tested in firefox and vivaldi)

**Comment by jhuda1** [comment]
> > Hi, did you used https for the page ? copy & paste requires it, in addition to allow clipboard access when prompted
> 
> Yes I can confirm that this is working with https and real certificate (tested in firefox and vivaldi)

> Hi, did you used https for the page ? copy & paste requires it, in addition to allow clipboard access when prompted


I really thank you,
But maybe you can explain to me what the connection is between https and copying and pasting from the clipboard?

**Comment by support-tt** [comment]
> But maybe you can explain to me what the connection is between https and copying and pasting from the clipboard?

The point with clipboard over the browser is, that this is security critical because a bad website could read your clipboard password etc. Thats why the browser companys decided the only https is alowed and you need to accept a popup from the browser. Some browser also want a Real certificate so you need lets encrypt or buy a cert. Unfortenetely you cant get around this (I tried :D ). Guacamole another Web based RDP/VNC solution in Linux has the same problem.

Sorry for my english .. :D

**Comment by jhuda1** [comment]
> > But maybe you can explain to me what the connection is between https and copying and pasting from the clipboard?
> 
> The point with clipboard over the browser is, that this is security critical because a bad website could read your clipboard password etc. Thats why the browser companys decided the only https is alowed and you need to accept a popup from the browser. Some browser also want a Real certificate so you need lets encrypt or buy a cert. Unfortenetely you cant get around this (I tried :D ). Guacamole another Web based RDP/VNC solution in Linux has the same problem.
> 
> Sorry for my english .. :D

It's strange, because I did what you say. I used an official certificate issued by lets encrypt (by ACME) and a window popped up asking me to confirm access to the board and I confirmed, but it didn't work. I will check it again.

And the English is fine, I don't speak English either...

**Comment by support-tt** [comment]
> > > But maybe you can explain to me what the connection is between https and copying and pasting from the clipboard?
> > 
> > 
> > The point with clipboard over the browser is, that this is security critical because a bad website could read your clipboard password etc. Thats why the browser companys decided the only https is alowed and you need to accept a popup from the browser. Some browser also want a Real certificate so you need lets encrypt or buy a cert. Unfortenetely you cant get around this (I tried :D ). Guacamole another Web based RDP/VNC solution in Linux has the same problem.
> > Sorry for my english .. :D
> 
> It's strange, because I did what you say. I used an official certificate issued by lets encrypt (by ACME) and a window popped up asking me to confirm access to the board and I confirmed, but it didn't work. I will check it again.
> 
> And the English is fine, I don't speak English either...

Hmm ok thats strange for me this is working. Maybe try another browser I had some issues with firefox which I dont have with google chrome.

**Comment by jhuda1** [comment]
So I was indeed able to use HTTPS to copy and paste using CTRL V, but not files from the desktop to the remote desktop. What could be the problem) @cedrozor @support-tt

**Comment by support-tt** [comment]
Great actually with copy paste this is also not working on my server but drag and drop is working.
Or with upload file on the bar on top (which in my case is the most solid way)
![image](https://user-images.githubusercontent.com/61587422/182358797-4c777eb6-39d0-4883-b2d0-afd807851491.png)

**Comment by jhuda1** [comment]
> מעולה למעשה עם העתק הדבק זה גם לא עובד על השרת שלי אבל גרור ושחרר עובד. או עם העלאת קובץ בסרגל למעלה (שבמקרה שלי היא הדרך הכי מוצקה) ![תמונה](https://user-images.githubusercontent.com/61587422/182358797-4c777eb6-39d0-4883-b2d0-afd807851491.png)

Ok, too bad it doesn't work by copy and paste maybe @cedrozor  will fix it in the next update of this amazing project.
I tried dragging anyway, it didn't really work...
Thank you very much you helped me
